test/functional/mempool_compatibility.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test that mempool.dat is both backward and forward compatible between versions
+
+NOTE: The test is designed to prevent cases when compatibility is broken accidentally.
+In case we need to break mempool compatibility we can continue to use the test by just bumping the version number.
+
+Download node binaries:
+contrib/devtools/previous_release.sh -b v0.19.1 v0.18.1 v0.17.1 v0.16.3 v0.15.2
+
+Only v0.15.2 is required by this test. The rest is used in other backwards compatibility tests.
+"""
+
+import os
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    adjust_bitcoin_conf_for_pre_17
+)
+
+class MempoolCompatibilityTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+        self.skip_if_no_previous_releases()
+
+    def setup_network(self):
+        self.add_nodes(self.num_nodes, versions=[
+            150200, # oldest version supported by the test framework
+            None,
+        ])
+        adjust_bitcoin_conf_for_pre_17(self.nodes[0].bitcoinconf)
```

Comment:
It looks like [other](https://github.com/bitcoin/bitcoin/blob/16d4b3fd6d5aad18ebb731a5006a15180d3661ef/test/functional/wallet_upgradewallet.py#L49) [instances](https://github.com/bitcoin/bitcoin/blob/16d4b3fd6d5aad18ebb731a5006a15180d3661ef/test/functional/feature_backwards_compatibility.py#L63) of `adjust_bitcoin_conf_for_pre_17` have left comments as to why it's used. I think it'd be helpful to include here as well.

Reply:
If you believe the code is unclear, we need to think of a better way to fix it rather than putting the same comment before each function call. For example: give this function a better name. But as for me, I think the function name speaks for itself and it doesn't need any clarifying comment.

Reply:
I think it's fine to call this without comment; it's always necessary for nodes before 0.17. Ideally the test framework (`add_nodes`) takes care of it automagically.